src/components/gltf-model-plus.js

```diff
@@ -407,6 +384,16 @@ AFRAME.registerComponent("gltf-model-plus", {
         if (el) rewires.push(() => (o.el = el));
       });
 
+      const environmentMapComponent = this.el.sceneEl.components["environment-map"];
+      const envMap = environmentMapComponent ? environmentMapComponent.environmentMap : null;
+
+      object3DToSet.traverse(object => {
+        if (object.material && object.material.isMeshStandardMaterial) {
+          object.material.envMap = envMap;
+          object.material.needsUpdate = true;
+        }
+      });
```

Comment:
I would have made a function on the environment map component called `apply` which applies the current environment map to an object hierarchy, and you could call it both here and in `updateEnvironmentMap`.